sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -134,16 +136,49 @@ def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
         - custom model name (``alias``) - used from ``algorithm_name``
         """
 
+    def _get_valid_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get valid parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            valid list of arguments for the underlying algorithm class
+        """
+        from pytorch_lightning import Trainer
+
+        model_class = self.algorithm_class
+        trainer_params = list(signature(Trainer.__init__).parameters.keys())
+        valid_parameters = list(signature(model_class.__init__).parameters.keys())
+        valid_parameters += trainer_params
+
+        sktime_parameters = self.algorithm_parameters
+
+        for sktime_param in list(sktime_parameters.keys()):
+            if sktime_param not in valid_parameters:
+                if sktime_parameters[sktime_param] is not None:
+                    warn(
+                        f"Keyword argument '{sktime_param}' will be omitted as it is"
+                        f" not found in the __init__ method "
+                        f"from {self.algorithm_class}. "
+                        f"Check your neuralforecast version"
+                        f"to find out the right API parameters."
+                    )
+                sktime_parameters.pop(sktime_param)
```

Comment:
I'll suggest to avoid using pop as it's an in-place modification and moat liky modify self.algorithm_parameters too. Will it make sense to do a deep copy in a new variable first, or maintain a separate dictionary with only valid parameters?

Reply:
yes I think it would be good to make a deepcopy of filter params